components/src/dynamo/planner/plugins/registry/gateway.py

```
@@ -0,0 +1,234 @@
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"heartbeat: malformed request: {type(exc).__name__}: {exc}",
+            )
+            raise  # unreachable: context.abort() raises AbortError
+        ok, reject = await self._server.authenticated_heartbeat(
+            pyd_req.plugin_id, pyd_req.auth_token
+        )
+        if reject == "auth_failed":
+            await context.abort(
+                grpc.StatusCode.UNAUTHENTICATED, "heartbeat: auth_failed"
+            )
+            raise  # unreachable
```

Ruff (0.15.14)

[error] Bare `raise` statement is not inside an exception handler

(PLE0704)